### PR TITLE
Make Debug Bar available to all proxied requests

### PR DESCRIPTION
**Pull request by mjangda**
- [x] Need to send no-cache headers
- [ ] Make it work without admin-bar
- [ ] Test, test, test

### CODE REVIEW

**Comment by joshbetz** [comment]
We should probably send nocache headers to avoid serving Debug Bar to all users.

**Comment by joshbetz** [comment]
Should we also enable debug bar for all logged in admins? I've previously thought about adding a `developer` role to enable tools like this for client devs, but not sure how we'd manage that.

File: debug-bar.php [hunk 1]
```diff
@@ -7,13 +7,20 @@
...
 		return true;
 	}
 	return $enable;
 }, 99 );
 
+add_action( 'send_headers', function() {
+	if ( true === A8C_PROXIED_REQUEST ) {
+		header( 'Cache-Control: private' );
+	}
+});
```
Comment (mjangda) [inline debug-bar.php]: Might be good to call `nocache_headers()` as well.

**Comment by mjangda** [comment]
Do we need to do something at the Varnish level? Proxied requests may end up getting a cached page and not see the debug bar.

File: debug-bar.php (hunk 1 above)
Comment (joshbetz) [inline debug-bar.php]: Oh yeah, that's definitely better.

**Comment by mjangda** [comment]
Bug: If a user is not logged in, the admin bar isn't shown. This means that the Debug doesn't render.

**Comment by mjangda** [comment]
> If a user is not logged in, the admin bar isn't shown. This means that the Debug Bar doesn't render.

On WP.com, the `debug-bar` node is removed from `$wp_admin_bar`. A new "super admin bar" is created that floats on the page, which might be the right direction for this.

**Comment by joshbetz** [comment]
What if, instead of making it work without the admin bar, we showed the admin bar to logged out, proxied users?

I think we can `add_filter( 'show_admin_bar', '__return_true' );`

**Comment by mjangda** [comment]
Sounds like a good workaround for now.

**Comment by simonwheatley** [comment]
> We should probably send nocache headers to avoid serving Debug Bar to all users.

Is this a good idea? It seems that it would make debugging caching issues harder. Should/could we have it off by default, and enable some toggle to turn it on (or vice versa)?

**Comment by mjangda** [comment]
I think as discussed in #602, it's probably worthwhile adding a button or something to the little sandbox bar/label that allows you to toggle uncached mode which could also load the Debug Bar. 

Not sure if it's useful to also stay in cached mode but still access the debug bar.

**Comment by mjangda** [comment]
Closing; we have Debug Mode now.

**Pull request closed without merging**